xmtp_mls/src/storage/encrypted_store/group.rs

```diff
@@ -185,13 +185,34 @@ impl DbConnection {
         &self,
         welcome_id: i64,
     ) -> Result<Option<StoredGroup>, StorageError> {
-        let mut query = dsl::groups.order(dsl::created_at_ns.asc()).into_boxed();
-        query = query.filter(dsl::welcome_id.eq(welcome_id));
+        let query = dsl::groups
+            .order(dsl::created_at_ns.asc())
+            .filter(dsl::welcome_id.eq(welcome_id));
+
         let groups: Vec<StoredGroup> = self.raw_query(|conn| query.load(conn))?;
         if groups.len() > 1 {
             tracing::error!("More than one group found for welcome_id {}", welcome_id);
         }
-        // Manually extract the first element
+
+        Ok(groups.into_iter().next())
+    }
+
+    pub fn find_dm_group(
+        &self,
+        dm_target_inbox_id: &str,
+    ) -> Result<Option<StoredGroup>, StorageError> {
+        let query = dsl::groups
+            .order(dsl::created_at_ns.asc())
+            .filter(dsl::dm_inbox_id.eq(Some(&dm_target_inbox_id)));
+
+        let groups: Vec<StoredGroup> = self.raw_query(|conn| query.load(conn))?;
+        if groups.len() > 1 {
+            tracing::error!(
+                "More than one group found for dm_inbox_id {}",
+                dm_target_inbox_id
+            );
+        }
```

Comment:
So I think this is likely going to happen frequently. With the way installations work you can over time get lots of dms with the same person. We are going to implement conversation threading eventually to make all these dm groups look like just one conversation from the user perspective. So I would just select the first or last? Group and not error. cc @insipx You can read more about this issue here https://github.com/xmtp/libxmtp/issues/696

Reply:
Got it, I'll downgrade this to info!.